### ISSUE TITLE
Set minimum stack size using PT_GNU_STACK header for Musl

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtimelab#791]
See http://git.musl-libc.org/cgit/musl/commit/?id=7b3348a98c139b4b4238384e52d4b0eb237e4833

This can be done once the minimum target musl version is 1.21. We are on 1.20 right now (it is what CI uses at least). 

Cleanup with this issue number once implemented.

**Comment by kant2002** [comment]
I have a question. Right now docker images for .NET Runtime seems to be shipping with version of Alpine at least 3.13
I take that from https://hub.docker.com/_/microsoft-dotnet-nightly-runtime/ 
According to Alpine website [version 3.13](https://alpinelinux.org/posts/Alpine-3.13.0-released.html) has Musl at least 1.2 which definitely bigger then 1.1.20, and even 3.10 (released 2019-06-13) has 1.1.22
Infra still uses 3.9 at least Linux-musl run on mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.9-WithNode-20200602002639-0fc54a3
2 years is probably too much to drop support for 3.9, but given that all containers on 3.13 I have mixed signals. It appears that only infra holding back, isn't it?

Second question. I notice following lines in `pal/src/CMakeList.txt`
https://github.com/dotnet/runtimelab/blob/5c2866363ebbd5364a4778d981cb61eaada039e3/src/coreclr/pal/src/CMakeLists.txt#L86-L92
In addition to `ENSURE_PRIMARY_STACK_SIZE` I see `DONT_SET_RLIMIT_NOFILE`, which does not present in `nativeaot` component. Should in't be in lines below
https://github.com/dotnet/runtimelab/blob/5c2866363ebbd5364a4778d981cb61eaada039e3/src/coreclr/nativeaot/CMakeLists.txt#L39-L43

**Comment by jkotas** [comment]
> It appears that only infra holding back, isn't it?

Yes, I have opened issue in dotnet/runtime to bump the alpine version used for builds.

> `DONT_SET_RLIMIT_NOFILE`

NativeAOT debugging is nothing like CoreCLR debugging. It is unlikely that whatever the issue is with CoreCLR debugging applies to NativeAOT too.
